src/test/FreeC/Pass/ExportPassTests.hs

```diff
@@ -0,0 +1,94 @@
+-- | This module contains tests for "FreeC.Pass.ExportPass".
+
+module FreeC.Pass.ExportPassTests
+  ( testExportPass
+  )
+where
+
+import           Test.Hspec
+
+import           Data.Maybe                     ( fromJust )
+import qualified Data.Set                      as Set
+import qualified Data.Text                     as Text
+
+import qualified FreeC.Backend.Coq.Syntax      as Coq
+import           FreeC.Environment
+import           FreeC.Environment.Entry
+import           FreeC.Environment.ModuleInterface
+import           FreeC.Monad.Class.Testable
+import           FreeC.Monad.Converter
+import           FreeC.Pass.ExportPass
+import           FreeC.Pretty
+import           FreeC.Test.Environment
+import           FreeC.Test.Parser
+
+
+shouldBeQualifiedWith :: Coq.Qualid -> String -> Converter Expectation
+shouldBeQualifiedWith qualid modName = do
+  case qualid of
+    Coq.Qualified s _ -> return $ Text.unpack s `shouldBe` modName
+    Coq.Bare _ ->
+      return
+        $    expectationFailure
+        $    showPretty
+        $    prettyString "Expected qualifier"
+        <$$> line
+        <>   indent 2 (prettyString modName)
+        <>   line
+        <$$> prettyString "but it was not qualified"
+
+testExportPass :: SpecWith ()
+testExportPass = describe "FreeC.Pass.ExportPass" $ do
+  context "Exported entries are qualified when exported " $ do
+    it "Names of data types should be qualified when exported" $ do
+      input <- expectParseTestModule ["module A where", "data Foo = Bar"]
```

Comment:
An input module of this form would not cause the environment to contain qualified names. Thus, the declarations should be qualified in this module, too.

```suggestion
      input <- expectParseTestModule ["module A where", "data A.Foo = A.Bar"]
```

In theory, it would also be possible to pass an empty module to the `exportPass`.

```suggestion
      input <- expectParseTestModule ["module A where"]
```

I'm not sure which option I like better. The nice thing about the latter option is that it sets the additional expectation that all information other than the module name is extracted from the environment. However, the former provides additional documentation and reflects the actual state of the module better.